utilities/component-statuses/componentStatusGenerator.js

```diff
@@ -0,0 +1,107 @@
+class ComponentStatusGenerator {
+    constructor ({
+        fs, path, promisify, console,
+    }) {
+        this.fs = fs;
+        this.path = path;
+        this.promisify = promisify;
+        this.console = console;
+
+        this.readdir = promisify(fs.readdir);
+        this.stat = promisify(fs.stat);
+        this.readFile = promisify(fs.readFile);
+        this.writeFile = promisify(fs.writeFile);
+
+        // These are packages that live in the components folder that we don't need to query for statuses
+        this.excludes = [
+            '@justeattakeaway/pie-webc',
+            '@justeattakeaway/pie-webc-core',
+            '@justeattakeaway/pie-webc-testing'
+        ];
+    }
+
+    // Finds the root of the monorepo by looking for a 'packages/components' directory
+    findMonorepoRootSync (startDir) {
+        let currentDir = startDir;
+
+        while (currentDir !== this.path.parse(currentDir).root) {
+            const potentialComponentsDir = this.path.join(currentDir, 'packages', 'components');
+            if (this.fs.existsSync(potentialComponentsDir) && this.fs.lstatSync(potentialComponentsDir).isDirectory()) {
+                return currentDir;
+            }
+            currentDir = this.path.dirname(currentDir);
+        }
+
+        throw new Error('Could not find the monorepo root.');
+    }
+
+    // Generates a json file with the statuses of each PIE component. The status is sourced from the package.json file of each component.
+    // The json will be outputted to the root of the package invoking this script as 'component-statuses.json'
+    async generateStatuses () {
+        this.console.info('Generating component statuses...');
+
+        try {
+            const monorepoRoot = this.findMonorepoRootSync(process.cwd());
```

Comment:
I think it will be easier to write tests for this function if the `workingDir` is a parameter instead, and then it can be passed in inside `generate-component-statuses.js`.